src/main/java/nl/tudelft/contextproject/saveLoad/LoadScript.java

```diff
@@ -0,0 +1,366 @@
+                } else {
+                    throw new XMLStreamException("Unexpected start tag in shot section: "
+                            + start.getName().getLocalPart());
+                }
+            } else if (event.isEndElement()) {
+                EndElement end = event.asEndElement();
+                if ("shots".equals(end.getName().getLocalPart())) {
+                    break;
+                } else {
+                    throw new XMLStreamException("Unexpected end tag in shots section: "
+                            + end.getName().getLocalPart());
+                }
+            }
+        }
+        return shots;
+    }
+
+    /**
+     * Reads a 'shot' section of the XML file and returns this as a Shot object.
+     * Assumes that the start element of this section has already been read and
+     * is inserted into this method as the startShot parameter.
+     * @return the loaded shot.
+     * @throws XMLStreamException when an error occurs in the XML.
+     */
+    private static Shot loadShot(StartElement startShot) throws XMLStreamException {
+        int id = Integer.parseInt(startShot.getAttributeByName(new QName("number")).getValue());
+        String shotId = "";
+        int cameraId = -1;
+        int presetId = -1;
+        String description = "";
+        while (reader.hasNext()) {
+            XMLEvent event = reader.nextEvent();
+            if (event.isStartElement()) {
+                StartElement start = event.asStartElement();
+                if ("shotId".equals(start.getName().getLocalPart())) {
+                    XMLEvent shotIdEvent = reader.nextEvent();
+                    if (shotIdEvent.isCharacters()) {
+                        shotId = shotIdEvent.asCharacters().getData();
+                    }
+                } else if ("description".equals(start.getName().getLocalPart())) {
+                    XMLEvent descEvent = reader.nextEvent();
+                    if (descEvent.isCharacters()) {
+                        description = descEvent.asCharacters().getData();
+                    }
+                } else if ("cameraId".equals(start.getName().getLocalPart())) {
+                    XMLEvent cameraIdEvent = reader.nextEvent();
+                    if (cameraIdEvent.isCharacters()) {
+                        cameraId = Integer.parseInt(cameraIdEvent.asCharacters().getData());
+                    } else {
+                        throw new XMLStreamException("No camera id present in shot.");
+                    }
+                } else if ("presetId".equals(start.getName().getLocalPart())) {
+                    XMLEvent presetIdEvent = reader.nextEvent();
+                    if (presetIdEvent.isCharacters()) {
+                        presetId = Integer.parseInt(presetIdEvent.asCharacters().getData());
+                    } else {
+                        throw new XMLStreamException("No preset id present in shot.");
+                    }
+                }
+            }
```

Comment:
You can probably put this if statement in another method. The current method is already too long.